### ISSUE TITLE
combine latest doesn't work the way (I think :-O) it should

### ISSUE THREAD

**Issue by BlueSolei** [Reactive-Extensions/RxCpp#395]
I am trying to use combine_latest() to transform a stream of points, to a stream of deltas.
The mouse is stream of stream of points - each stream start on mouse down and ends on mouse up.
Attached here my simple test.
The expected output is:
  Delta no 0 is: 0,0
  Delta no 1 is: 5,0
  Delta no 2 is: 0,5
  Delta no 3 is: 2,3

But  I only get the first line:
  Delta no 0 is: 0,0

```cpp
#include <rx.hpp>
namespace rx = rxcpp;
namespace rxsub = rxcpp::subjects;
using rxob = rx::observable<>;

struct Point
{
	Point(int x, int y) : x(x), y(y) {}

	int x = 0, y = 0;
	Point operator-() const { return {-x, -y}; }
	Point operator+(const Point& other) const { return Point{x + other.x, y + other.y}; }
	Point operator-(const Point& other) const { return operator+(-other); }
};

std::ostream& operator<<(std::ostream& o, const Point& p)
{
	return o << "(" << p.x << ", " << p.y << ")";
}

void Test()
{
	using RxPoint = rx::observable<Point>;
	using Strokes = rx::observable<RxPoint>;
	using StrokesSubject = rxsub::subject<RxPoint>;

	StrokesSubject strokesSource;
	auto strokes = strokesSource.get_observable();

	auto deltas = [](Strokes strokes) {
		auto d = strokes.flat_map([=](RxPoint stroke) {
			auto points = stroke;
			// create stream of delta vectors from start point
			auto firstPoint = points.take(1);
			auto delta =
				points.combine_latest([](Point v0, Point v1) { return v0 - v1; }, firstPoint);
			return delta;
		});

		return d;
	};

	auto delta = deltas(strokes);
	int n = 0;
	delta.subscribe(
		[&](const Point& d) { std::cout << "Delta no. " << n++ << " is: " << d << std::endl; });

	auto testMouse = rxob::from(Point{3 + 0, 4 + 0}, Point{3 + 5, 4 + 0}, Point{3 + 0, 4 + 5},
								Point{3 + 2, 4 + 3});
	strokesSource.get_subscriber().on_next(testMouse);
}

```

**Comment by kirkshoop** [comment]
this is a HOTvCOLD behavior. 

the strokes are COLD and are being shared and only one thread is used. the `points.combine_latest(..., firstPoint)` means that all the `points` are sent before `firstPoint` is subscribed. thus only the last delta is emitted.

COLD and HOT sources will work if you reverse the `combine_latest`

```cpp
			auto delta =
				firstPoint.combine_latest([](Point v0, Point v1) { return v1 - v0; }, points);
```

**Comment by BlueSolei** [comment]
Thank you very much!

**Comment by BlueSolei** [comment]
I tried to make the observables (strokes & testMouse) hot, so my original example will work, but the result is no output at all.
I changed the example to:
```cpp
auto strokes = strokesSource.get_observable().publish().ref_count();
```
```cpp
auto testMouse = rxob::from(Point{3 + 0, 4 + 0}, Point{3 + 5, 4 + 0}, Point{3 + 0, 4 + 5},
						Point{3 + 2, 4 + 3}).publish().ref_count();
```
Can you help me understand why is that?

**Comment by kirkshoop** [comment]
> the first delta which is emitted "Delta no 0 is (0,0)", not the last one

When I ran the original code I never saw that outcome. I saw
```
Delta no. 0 is: (2, 3)
```
perhaps there is some other code in your app that is affecting the outcome or perhaps you are using a different commit of rxcpp than I am?

> the result is no output at all 

yes, I saw that too. :)
The issues is that all the items from teh hot stream are added before the second subscription to teh hot stream which gets none.
